cmd/node_agent/na/aws.go

```diff
@@ -41,16 +68,44 @@ func (na *awsPlatformImpl) IsProperPlatform() bool {
 	return na.client.Available()
 }
 
-// Extract service identity from userdata. This function should be
-// pluggable for different AWS deployments in the future.
+// This function should be pluggable for different AWS deployments.
 func (na *awsPlatformImpl) GetServiceIdentity() (string, error) {
 	return "", nil
 }
 
+func (na *awsPlatformImpl) GetInstanceIdentityDocument() ([]byte, error) {
+	cert, err := pki.ParsePemEncodedCertificate([]byte(AWSCertificatePem))
+	if err != nil {
+		return []byte{}, fmt.Errorf("Failed to parse AWS public certificate: %v", err)
```

Comment:
what about return nil instead of []byte{} ?

Reply:
same below

Reply:
Done.